### PR TITLE
Add automated tests for complex workflows around creating a prefab

### PR DESCRIPTION
**Pull request by chiyenteng**
Add the following new automated tests for complex workflows around creating a prefab:
- CreatePrefab_UnderChildEntityOfAnotherPrefab
- CreatePrefab_WithNestedEntities
- CreatePrefab_WithNestedEntitiesAndNestedPrefabs

Tested with the following commands:
`python\python.cmd -m pytest AutomatedTesting\Gem\PythonTests\Prefab\TestSuite_Main.py --build-directory windows_vs2019\bin\profile`
`python\python.cmd -m pytest AutomatedTesting\Gem\PythonTests\Prefab\TestSuite_Main_Optimized.py::TestAutomationNoAutoTestMode --build-directory windows_vs2019\bin\profile`

Signed-off-by: chiyenteng <82238204+chiyenteng@users.noreply.github.com>

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py [hunk 1]
```diff
@@ -18,6 +19,110 @@
 import azlmbr.entity as entity
 import azlmbr.legacy.general as general
 
+def get_linear_nested_items_name(nested_items_name_prefix, current_layer):
```
Comment (AMZN-daimini) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py:22]: I found the use of the word "layer" confusing in this context, do you mean level?

Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py:22]: Agreed, think "level" in regards to hierarchies makes a bit more sense than layer.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py [hunk 2]
```diff
@@ -18,6 +19,110 @@
...
+    return f"{nested_items_name_prefix}{current_layer}"
+
+def create_linear_nested_entities(nested_entities_name_prefix, layer_count, pos):
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py:25]: I think docstrings would be helpful for these helpers.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+"""
+
+def CreatePrefab_UnderChildEntityOfAnotherPrefab():
+    """
+    Test description:
+    - Creates parent/child entities with the entities having a physx collider
+    - Creates a prefab "Outer_prefab" and an instance based of the entities
+    - Creates a prefab "Inner_prefab" inside "Outer_prefab" based the child entity contained inside of it
+
+    Checks that the parent/child entity is correctly handlded by the prefab system checking the name and that it contains the physx collider
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py:15]: Small typo with "handlded"

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    PHYSX_COLLIDER_NAME = "PhysX Collider"
+
+    prefab_test_utils.open_base_tests_level()
+
+    # Creates new parent/child Entities at the root level
+    # Asserts if creation didn't succeed
+    parent_entity = EditorEntity.create_editor_entity_at((100.0, 100.0, 100.0), name=PARENT_ENTITY_NAME)
+    assert parent_entity.id.IsValid(), "Couldn't create parententity"
+    parent_entity.add_component(PHYSX_COLLIDER_NAME)
+    assert parent_entity.has_component(PHYSX_COLLIDER_NAME), "Attempted to add a PhysX Collider but no physx collider collider was found afterwards"
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py:40]: I'd simplify this assert statement to something like "Failed to add a PhysX Collider"

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    PARENT_ENTITY_NAME = "ParentEntity"
+    CHILD_ENTITY_NAME = "ChildEntity"
+    PHYSX_COLLIDER_NAME = "PhysX Collider"
+
+    prefab_test_utils.open_base_tests_level()
+
+    # Creates new parent/child Entities at the root level
+    # Asserts if creation didn't succeed
+    parent_entity = EditorEntity.create_editor_entity_at((100.0, 100.0, 100.0), name=PARENT_ENTITY_NAME)
+    assert parent_entity.id.IsValid(), "Couldn't create parententity"
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py:38]: Missing space between parent and entity.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_WithNestedEntitiesAndNestedPrefabs.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+"""
+
+def CreatePrefab_WithNestedEntitiesAndNestedPrefabs():
+    """
+    Test description:
+    - Creates linear nested entities. 
+    - Creates a prefab from the nested entities.
+
+    Test passed if the newly instanced prefab of the 3 entities have correct hierarchy and positions.
+    """
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_WithNestedEntitiesAndNestedPrefabs.py:15]: This test has the exact same test description as CreatePrefab_WithNestedEntities

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_WithNestedEntitiesAndNestedPrefabs.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    import Prefab.tests.PrefabTestUtils as prefab_test_utils
+
+    NESTED_ENTITIES_PREFAB_FILE_NAME = Path(__file__).stem + '_' + 'nested_entities_prefab'
+    NESTED_ENTITIES_NAME_PREFIX = 'Entity_'
+    NESTED_PREFABS_FILE_NAME_PREFIX = Path(__file__).stem + '_' + 'nested_prefabs_'
+    NESTED_PREFABS_NAME_PREFIX = 'NestedPrefabs_Prefab_'
+    FILE_NAME_OF_PREFAB_WITH_NESTED_ENTITIES_AND_NESTED_PREFABS = Path(__file__).stem + '_' + 'new_prefab'
+    NESTED_PREFABS_TEST_ENTITY_NAME = 'TestEntity'
+    PHYSX_COLLIDER_NAME = "PhysX Collider"
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_WithNestedEntitiesAndNestedPrefabs.py:30]: NIT: Double/Single quote usage isn't consistent.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py [hunk 8]
```diff
@@ -19,38 +19,114 @@
...
+
+        current_entity = child_entity
+        
+    assert level_count == expected_level_count, \
+        f"Number of levels of nested entities should be {expected_level_count}, not {level_count}"
+
+def create_linear_nested_prefabs(entities, nested_prefabs_file_name_prefix, nested_prefabs_instance_name_prefix, level_count):
+    """
+    This is a helper function which helps create nested prefabs 
+    where each nested prefab has only one child prefab most. For example:
```
Comment (jckand-amzn) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py:94]: Missing "at" here for "at most".

**Review by jckand-amzn — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py (hunk 8 above)
Comment (chiyenteng) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/PrefabTestUtils.py]: Fixed.

**Review by AMZN-daimini — APPROVED** [review]

**Review by AMZN-mnaumov — APPROVED** [review]

**Pull request merged**